### PR TITLE
Add EnablePartitioning to QueueProperties ctor; rename GetRawMessage

### LINKED ISSUE #17100: Port ATOM XML fix to Azure.Messaging.ServiceBus
We should take the fixes from https://github.com/Azure/azure-sdk-for-net/pull/16488 and https://github.com/Azure/azure-sdk-for-net/pull/16526

### PR DESCRIPTION
**Pull request by JoshLove-msft**
- EnablePartitioning was missing in the QueueProperties ctor
- Renaming GetRawMessage to GetRawAmqpMessage
- Porting ATOM fix https://github.com/Azure/azure-sdk-for-net/issues/17100

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/QueueProperties.cs [hunk 1]
```diff
@@ -327,6 +347,9 @@ public bool Equals(QueueProperties other)
                 && RequiresDuplicateDetection.Equals(otherDescription.RequiresDuplicateDetection)
                 && RequiresSession.Equals(otherDescription.RequiresSession)
                 && Status.Equals(otherDescription.Status)
+                && _internalSupportOrdering.Equals(other.SupportOrdering)
```
Comment (yvgopal) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/QueueProperties.cs]: Same issue you pointed out.

**Comment by yvgopal** [comment]
>         internal List<object> UnknownProperties { get; set; }

Type. You are changing it to XElement in QueueProperties. May be you can keep the type consistent.

---
Refers to: sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/TopicProperties.cs:209 in 9a1ffe0. [](commit_id = 9a1ffe0d6c90b3e4e5994e824e44c726281512f7, deletion_comment = False)

**Review by yvgopal — APPROVED** [review]
Queue equals has the same issue you flagged in Track 1 SDK.
Nit: Unknown elements type is not consistent. Stick with List<Object> or List<XElement> in all 3 properties classes.

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/QueuePropertiesExtensions.cs [hunk 2]
```diff
@@ -107,65 +112,73 @@ private static async Task<QueueProperties> ParseFromEntryElementAsync(XElement x
...
-                        properties.Status = element.Value;
-                        break;
-                    case "AutoDeleteOnIdle":
-                        properties.AutoDeleteOnIdle = XmlConvert.ToTimeSpan(element.Value);
+                    case "IsAnonymousAccessible":
+                        properties.IsAnonymousAccessible = Boolean.Parse(element.Value);
                         break;
-                    case "EnablePartitioning":
-                        properties.EnablePartitioning = bool.Parse(element.Value);
+                    case "AuthorizationRules":
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/QueuePropertiesExtensions.cs:133]: Not worth taking on this close to code complete, but may be worth a follow-up at some point to pull the magic strings out into a set of constants.

**Review by jsquire — APPROVED** [review]

**Pull request merged**